packages/tooltip/src/Tooltip.js

```diff
@@ -5,11 +5,18 @@ import values from "object.values";
 import { Container, StyledTooltip } from "./styledComponents/Tooltip";
 import alignments from "./constants/alignments";
 
-export const getAutoAlign = bodyWidth => containerRef => {
-  if (containerRef) {
-    const { x } = containerRef.getBoundingClientRect(); // Position of the element on viewport
-    const halfViewport = bodyWidth / 2;
-    if (x > halfViewport) return alignments.RIGHT;
+export const getAutoAlign = (tooltipRef, bondariesSelector) => {
+  if (tooltipRef) {
+    const { right } = tooltipRef.getBoundingClientRect();
+
+    const bondariesNode =
+      bondariesSelector && tooltipRef.closest(bondariesSelector);
+
+    const maxRight =
+      (bondariesNode && bondariesNode.getBoundingClientRect().right) ||
+      window.innerWidth;
+
+    if (right >= maxRight) return alignments.RIGHT;
```

Comment:
@alcferreira @fczuardi what you guys think of this approach?

We're getting the right border of the tooltip and comparing it to the given container right border.
If no `bondariesSelector` is given, the limit will be the viewport.

Reply:
Really nice! I like it better now